### PR TITLE
sjawhar-legion-373: strip file registry + update oc ps to use sessions API

### LINKED ISSUE #373: feat(envoy): consolidate file + KV session registries into KV-only
## Problem

The Envoy listener has two session registries (file + KV) that can disagree, causing stale-port delivery failures. `oc ps` is unreliable tech debt that scrapes local PID files and knows nothing about KV or remote sessions.

## Design: Pluggable registry

The `Deliverer` uses a `SessionLookup` interface. Two implementations, **never both at once**:

### File registry (`ENVOY_SESSION_REGISTRY=file`)
- Plugin writes `$OC_REGISTRY/<sessionID>.json`
- Listener reads from `ENVOY_REGISTRY_DIR` (volume-mounted)
- No KV session bucket
- For: single-machine, local dev, minimal infrastructure

### KV registry (`ENVOY_SESSION_REGISTRY=kv`, default)
- Plugin writes via HTTP to listener subscribe endpoint
- Listener reads from `envoy_sessions` KV
- No file registry, no volume mount
- For: production fleet, cross-machine delivery

**When KV is selected, the file path is dead code.** No fallback chain, no disagreements.

## Design: Replace `oc ps`

Two complementary approaches:

### Immediate: Standalone script (Option C)
- `~/.dotfiles/scripts/oc ps` calls `GET /v1/sessions` on the listener API
- Zero dependencies beyond curl
- Bridges until the Go CLI exists

### Long-term: Go CLI via Docker alias
```bash
alias envoy='docker run --rm -it --network host ${ENVOY_IMAGE}'
# Usage: envoy ps, envoy subscribe, envoy sessions
```
- Add CLI entrypoint to the existing Go binary (alongside listener, receivers)
- Talks to NATS directly from inside the container — no API endpoint needed
- `envoy ps` shows all sessions fleet-wide
- Could also expose `envoy subscribe`, `envoy unsubscribe`, `envoy inspect <session>` for debugging
- Zero host dependencies beyond Docker

## Migration

1. Add `SessionLookup` interface + config switch to listener
2. Add `GET /v1/sessions` endpoint (#370)
3. Update `oc ps` script to use the API
4. Add `envoy` CLI entrypoint to Go binary (Dockerfile already builds all binaries)
5. Strip dual-registry fallback from `Deliverer`
6. Clean up plugin file-writing code when KV is selected

## References
- Stale port debugging session that exposed the dual-registry problem
- #370 (sessions API endpoint)

### PR DESCRIPTION
**Pull request by sjawhar**
Closes #373

## Summary

Implements sub-items (3) and (5) from the session registry consolidation (#373). Sub-issues #375 (SessionLookup interface) and #376 (GET /v1/sessions endpoint) were already merged.

### (3) Update `oc ps` to use sessions API
- Rewrote `_oc_ps()` to `curl GET $ENVOY_URL/v1/sessions` instead of scraping PID files from `$OC_REGISTRY`
- New columns: SESSION, PORT, MACHINE, DIR, TOPICS, UPDATED
- Removed dead file-writing code from `oc` main function (OC_REGISTRY, registry file, trap)
- **Note:** `oc` script lives in `~/.dotfiles/scripts/oc`, not in this repo

### (5) Strip dual-registry fallback + plugin file-writing cleanup

**Go (packages/envoy/)**
- Deleted `file_registry.go` + `file_registry_test.go`
- Simplified `lookup.go`: removed `NewSessionLookup`, `ParseRegistryMode`, mode constants. Kept `SessionLookup` interface for testability
- Updated `main.go`: removed `ENVOY_SESSION_REGISTRY`/`ENVOY_REGISTRY_DIR` config, use `OpenSessionRegistry` directly
- Removed dead `RegistryEntry` struct from `session.go`
- Converted all file-based tests to KV-backed via NATS testcontainers
- Shared NATS container across tests (sync.Once) for faster test runs
- Fixed bucket cleanup to delete-and-recreate for TTL-sensitive tests

**TypeScript (packages/envoy-plugin/)**
- Removed file-writing: `readFileSync`/`writeFileSync`/`unlinkSync`, `registryDir`, `update()`, `registryFiles`
- Removed dead `fetchSession()` (only used for file metadata)
- Made heartbeat/port-sync unconditional (no longer gated on `registryDir`)
- Simplified event handler: only `session.status/busy` for Envoy subscribe

**Infra/deploy cleanup**
- Removed `ENVOY_REGISTRY_DIR` from compose, Pulumi config, and infra types
- Removed registry volume mount from listener container

### CODE REVIEW

**Comment by sjawhar** [comment]
## Behavioral Test Results

### Summary
**PASS** — 5/5 verifiable acceptance criteria verified. 2 criteria are external to this repo (unverifiable from workspace, not failures).

### Results
| Criterion | Status | Evidence |
|-----------|--------|----------|
| **(5a)** Delete `file_registry.go`, remove `NewSessionLookup`/`ParseRegistryMode`/mode constants | ✅ | `ls packages/envoy/internal/session/file_registry*.go` returns nothing. `lookup.go` is 8 lines: only `SessionLookup` interface remains. Verified by code inspection: `packages/envoy/internal/session/lookup.go:1-8` |
| **(5b)** Remove `ENVOY_SESSION_REGISTRY`/`ENVOY_REGISTRY_DIR` from main.go, plugin file-writing, dead `RegistryEntry` struct | ✅ | `grep -r 'ENVOY_REGISTRY_DIR\|ENVOY_SESSION_REGISTRY\|NewSessionLookup\|ParseRegistryMode\|RegistryEntry' packages/envoy/internal/ packages/envoy/cmd/ packages/envoy-plugin/src/` — zero hits in production code. Plugin `index.ts` has no `readFileSync`/`writeFileSync`/`unlinkSync`/`registryDir`/`update()`/`fetchSession()`. |
| **(5c)** Heartbeat/port-sync unconditional, infra cleanup | ✅ | Plugin heartbeat timer runs unconditionally (not gated on `registryDir`). Compose file has no `ENVOY_REGISTRY_DIR` env or registry volume. Pulumi config and `services.ts` have no registry references. `machines.ts` `ListenerConfig` has no `registryDir` field. |
| **(5d)** Tests converted from file-based to KV-backed | ✅ | All 97 Go tests pass locally with real NATS testcontainers: `session/` (28 pass), `cmd/listener/` (22 pass), `mcpbridge/` (47 pass). Shared NATS container via `sync.Once`. CI: all 6 checks pass (envoy-go 4m36s, envoy-plugin 15s). |
| **(5e)** `SessionLookup` interface kept for testability | ✅ | Interface exists in `lookup.go`, used by `Deliverer.Sessions` field in `session.go`. Minor drift: `listenerDeps.sessions` in `main.go` is concrete `*SessionRegistry` (needed for `List()` on `/v1/sessions`). |
| **(3a)** `oc ps` calls `GET /v1/sessions` instead of scraping PID files | N/A | Script lives in `~/.dotfiles/scripts/oc`, outside this repo. Cannot verify from workspace. |
| **(3b)** Standalone script with zero deps beyond curl | N/A | Same — external to this repo. |

### Test Quality Critique
- **Positive: Real integration tests.** Session/delivery/handler tests use real NATS testcontainers, not mocked KV. This catches actual JetStream behavior (TTL expiry, bucket lifecycle). Good alignment with the `nats-kv-testing-patterns` learning.
- **Positive: Good edge case coverage.** Tests cover nil receivers (`ErrNoKV`), wrong-machine delivery (ACK not NAK), dead sessions excluded from `/v1/sessions`, orphan sessions without interests.
- **Positive: Race condition fix is well-targeted.** The `http_server_test.go` fix (set SSE fields BEFORE writing endpoint event) addresses a real CI flake with clear comments explaining the root cause.
- **P2: Stale documentation references.** `packages/envoy/deploy/README.md` (lines 74, 83) still references `ENVOY_REGISTRY_DIR`. `packages/envoy/infra/README.md` (line 108) still documents `listener.registryDir` as a required field. These should be cleaned up.
- **P2: No negative test for file registry removal.** No test explicitly asserts the file-based registration path is gone (e.g., verifying `ENVOY_SESSION_REGISTRY=file` is rejected). This is tested implicitly by the absence of file-based test infrastructure, but an explicit rejection test would catch accidental reintroduction.

### Documentation Feedback
- **PR description is excellent.** Clear separation of sub-items (3) and (5), with specific file-level changes documented.
- **Implementation comment on the issue** provides a good summary with CI status.
- **Stale docs**: Two README files still reference removed `ENVOY_REGISTRY_DIR` and `listener.registryDir`. These would confuse anyone following the deployment docs.
- **No doc updates for the registry consolidation story.** The deploy README's "Listener envs" section should be updated to remove the registry dir reference and note that KV is the only registry mode.

### Observations
- The concrete `*SessionRegistry` in `listenerDeps` (instead of `SessionLookup` interface) is a pragmatic choice — `/v1/sessions` needs `List()` which isn't on the interface. Worth documenting if someone later wants to swap the registry implementation.
- Race condition fix in `http_server_test.go` is out of scope for the registry consolidation issue but is a legitimate test stability improvement.

### Config Compliance
- require_specific_task: false — N/A (default)
- require_taiga_evidence: false — N/A (default)

**Comment by sjawhar** [comment]
## Review Summary

**CRITICAL (P1):** 0 issues
**IMPORTANT (P2):** 2 issues
**MINOR (P3):** 2 suggestions

**Verdict: Approved to merge** — no blocking issues.

---

### Acceptance Criteria Verification

| Migration Step | Status | Evidence |
|---|---|---|
| (1) SessionLookup interface + config switch | ✅ Already merged (#375, #376) | `SessionLookup` interface retained in `lookup.go` |
| (2) GET /v1/sessions endpoint | ✅ Already merged (#370) | Handler in `main.go` uses `SessionRegistry` directly |
| (3) Update `oc ps` to use sessions API | ✅ External to repo | PR description notes `oc` script lives in `~/.dotfiles/scripts/oc` |
| (4) Go CLI via Docker alias | 🚫 Future work | Issue describes this as "long-term" |
| (5) Strip dual-registry fallback | ✅ Implemented | `Deliverer.Deliver()` is KV-only, `file_registry.go` deleted, no file fallback path remains |
| (6) Plugin file-writing cleanup | ✅ Implemented | All `readFileSync`/`writeFileSync`/`unlinkSync`, `registryDir`, `registryFiles`, `fetchSession()` removed |

### Key Findings

**P2: Stale README references to removed config**
- `packages/envoy/deploy/README.md` lines 74, 83: still lists `ENVOY_REGISTRY_DIR` as required env var
- `packages/envoy/infra/README.md` line 108: still documents `listener.registryDir` as required config
- These will mislead deployers following the docs. The tester also flagged this.

**P2: `envoy-go` CI check pending**
- 5/6 CI checks pass (contracts ✅, envoy-plugin ✅, lint ✅, typecheck ✅, test ✅)
- `envoy-go` still pending — this PR deletes Go files, refactors tests to shared NATS containers, and simplifies the registry. Go CI confirmation is important.

**P3: Missing trailing newline in `.legion/implement.json` and `.legion/test.json`**
- POSIX convention. Very minor.

**P3: `listenerDeps.sessions` uses concrete `*SessionRegistry` instead of `SessionLookup` interface**
- Pragmatic — needed for `List()` on `/v1/sessions` endpoint. Acceptable trade-off since there is now only one implementation.

### What Went Well

- **Thorough deletion**: File registry code, tests, infra config, plugin file-writing all cleanly removed. No stale references in code (only README docs).
- **Test conversion**: All tests properly converted from file-backed to KV-backed with shared NATS containers. Follows established `nats-kv-testing-patterns` (bucket delete-and-recreate for TTL-sensitive tests).
- **Race condition fix in `http_server_test.go`**: `sseWriter`/`sseConn` fields now set before writing the endpoint event, preventing the client from sending requests to nil fields. Correctly scoped to test mock only.
- **Plugin simplification**: Heartbeat/port-sync now unconditional (correct for KV-only world). Event handler properly simplified — only `session.status/busy` needed for Envoy subscribe.
- **Net -343 lines**: Clean removal of dead code paths.

### Evidence
- CI status: `contracts ✅ | envoy-plugin ✅ | lint ✅ | typecheck ✅ | test ✅ | envoy-go ⏳`
- Stale refs confirmed: `grep -n "ENVOY_REGISTRY_DIR\|registryDir" packages/envoy/deploy/README.md packages/envoy/infra/README.md` → 3 matches
- No stale Go refs: `grep -r "FileRegistry\|file_registry\|ENVOY_SESSION_REGISTRY\|NewSessionLookup\|ParseRegistryMode" packages/envoy/ --include="*.go"` → 0 matches (only test names like `TestDeliver_NoRegistryEntry` remain, which describe scenarios not the old code)
- No stale TS refs: `grep -r "OC_REGISTRY\|registryDir\|registryFile\|readFileSync\|writeFileSync\|unlinkSync\|fetchSession" packages/envoy-plugin/ --include="*.ts"` → 0 matches

File: packages/envoy/cmd/listener/main.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ import (
 type listenerDeps struct {
 	client   *bus.Client
 	registry *store.Registry
-	sessions session.SessionLookup
+	sessions *session.SessionRegistry
```
Comment (sjawhar) [inline packages/envoy/cmd/listener/main.go]: **[P3]:** `listenerDeps.sessions` changed from `SessionLookup` interface to concrete `*SessionRegistry`. This is pragmatic — `List()` is needed for `/v1/sessions` and isn't on the interface — but it does reduce testability of the listener init code. Acceptable since there's now only one implementation.

File: packages/envoy/internal/mcpbridge/http_server_test.go [hunk 2]
```diff
@@ -44,15 +44,19 @@ func newMockHTTPMCPServer(opts ...func(*mockHTTPMCPServer)) *mockHTTPMCPServer {
...
 		w.WriteHeader(http.StatusOK)
 
-		messageURL := "http://" + r.Host + "/message"
-		fmt.Fprintf(w, "event: endpoint\ndata: %s\n\n", messageURL)
-		flusher.Flush()
-
+		// Set SSE fields BEFORE writing the endpoint event so they're
+		// available when the client reads the event and sends requests.
+		// Otherwise there's a race: the client reads the endpoint event,
+		// immediately POSTs to /message, and sendSSE() finds nil fields.
```
Comment (sjawhar) [inline packages/envoy/internal/mcpbridge/http_server_test.go]: **[P3 — Positive]:** Good catch on the race condition. Moving `sseWriter`/`sseConn` assignment before the endpoint event write correctly prevents the race where the client reads the event, immediately POSTs to `/message`, and `sendSSE()` finds nil fields. The comment explains the rationale clearly.
